code/modules/pulsar_engine/pulsar_consoles.dm

```diff
@@ -0,0 +1,269 @@
+	connect_to_network()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/power/pulsar_power_bridge/LateInitialize()
+	. = ..()
+	pulsar_console = locate() in world //I can get away with it once, right?
+	if(pulsar_console)
+		console_area = get_area(pulsar_console) //Area stored so reconnections are cheaper.
+
+/obj/machinery/power/pulsar_power_bridge/Process()
+	if(powernet && pulsar_console)
+		add_avail(PULSAR_100_POWER * pulsar_console.get_effective_power_porduced())
+	. = ..()
+
+/obj/machinery/power/pulsar_power_bridge/power_change()
+	..()
+	SSnano.update_uis(src)
+
+/obj/machinery/power/pulsar_power_bridge/attack_hand(mob/user)
+	. = ..()
+	nano_ui_interact(user)
+
+/obj/machinery/power/pulsar_power_bridge/nano_ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = NANOUI_FOCUS)
+	. = ..()
+	if(stat & (BROKEN|NOPOWER)) return
+	if(user.stat || user.restrained()) return
+
+	var/list/data = ui_data()
+
+	ui = SSnano.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "pulsar_power_bridge.tmpl", name, 550, 400)
+		ui.set_initial_data(data)
+		ui.open()
+
+/obj/machinery/power/pulsar_power_bridge/ui_data()
+	var/list/data = list()
+	data["pulsar_connected"] = FALSE
+	if(pulsar_console)
+		data["pulsar_connected"] = TRUE
+		data["power_percentage"] = pulsar_console.get_effective_power_porduced()
+		data["power_produced"] = PULSAR_100_POWER * pulsar_console.get_effective_power_porduced()
+
+	return data
+
+/obj/machinery/power/pulsar_power_bridge/Topic(href, href_list)
+	if(href_list["disconnect"])
+		pulsar_console = null
+	else if(href_list["reconnect"])
+		pulsar_console = locate() in console_area
+	SSnano.update_uis(src)
+
+/obj/machinery/power/pulsar_power_bridge/examine(mob/user, distance, infix, suffix)
+	..(user)
+	to_chat(user, "\The [src] appears to be producing [PULSAR_100_POWER * pulsar_console.get_effective_power_porduced()] W.")
+
+/obj/machinery/power/pulsar_power_bridge/Destroy()
+	pulsar_console = null
+	. = ..()
+
+/obj/structure/pulsar_fuel_tank
+	name = "pulsar fuel tank"
+	desc = "A massive fuel tank refialable by smaller gas tanks."
+	icon = 'icons/atmos/tank.dmi' //All sprites are placeholders
+	icon_state = "plasma_map"
+	anchored = TRUE
+
+	var/obj/machinery/pulsar/connected_console //Used to handle proper qdels, couldn't find a signal for it
+	var/datum/gas_mixture/air_contents
+	var/volume = 700
+
+/obj/structure/pulsar_fuel_tank/Initialize()
+	. = ..()
+	air_contents = new /datum/gas_mixture(volume)
+	air_contents.temperature = T20C
+	air_contents.adjust_gas("plasma", 3 * ONE_ATMOSPHERE * volume/(R_IDEAL_GAS_EQUATION*T20C))
+
+
+/obj/structure/pulsar_fuel_tank/Destroy()
+	. = ..()
+	if(air_contents)
+		QDEL_NULL(air_contents)
+	connected_console.tank = null
+	SSnano.update_uis(connected_console)
+
+/obj/structure/pulsar_fuel_tank/attackby(obj/item/W as obj, mob/user as mob)
+	if(istype(W,/obj/item/tank))
+		var/obj/item/tank/tank = W
```

Comment:
```suggestion
/obj/structure/pulsar_fuel_tank/attackby(obj/item/tank, mob/user)
	if(istype(tank))
```

Reply:
Should work with `obj/item/tank/tank` as istype tank will take any obj/item

Reply:
Touché.